File: app/src/main/java/com/parallelia/gustavo/parallelia/model/KNN_Vector.java
package com.parallelia.gustavo.parallelia.model;

import android.graphics.Bitmap;

/**
 * Created by gustavo on 16/01/15.
 */
public class KNN_Vector {
    //atributos
    private float label;
    private int[] eigenvector;

    public KNN_Vector(Bitmap image, float label){
        this.label = label;
        this.eigenvector = new int[image.getWidth()*image.getHeight()];
        //copy pixel data from the Bitmap into the 'intArray' array
        image.getPixels(this.eigenvector, 0, image.getWidth(), 0, 0, image.getWidth(), image.getHeight());
        //this.setEigenvector(eigenvector);
    }


    public float getLabel() {
        return label;
    }

    public void setLabel(float label) {
        this.label = label;
    }

    public int[] getEigenvector() {
        return eigenvector;
    }

    public void setEigenvector(Bitmap image) {
        this.eigenvector = new int[image.getWidth()*image.getHeight()];
        //copy pixel data from the Bitmap into the 'intArray' array
        image.getPixels(this.eigenvector, 0, image.getWidth(), 0, 0, image.getWidth(), image.getHeight());
        //this.eigenvector = eigenvector;
    }
}

File: PC - SBC/KNN/KNN OpenMP/KNNVector.cpp
#include "opencv2/opencv.hpp"
#include <iostream>
#include <stdio.h>
#include "KNNVector.hpp"

KNNVector::KNNVector() {
	Mat infoLetter;
	letterLabel = 0;
}

KNNVector::KNNVector(Mat letter, float label) {
    letterLabel = label;
    infoLetter = letter;
}


float KNNVector::getLabel() {
    return letterLabel;
}

void KNNVector::setLabel(float label) {
    letterLabel = label;
}

Mat KNNVector::getEigenvector() {
    return infoLetter;
}

void KNNVector::setEigenvector(Mat letter) {
	infoLetter = letter;
}